server/src/main/java/io/crate/fdw/JdbcForeignDataWrapper.java

```diff
@@ -144,9 +146,12 @@ public CompletableFuture<BatchIterator<Row>> getIterator(Role currentUser,
 
         Settings options = server.options();
         String url = urlSetting.get(options);
+        if (!url.startsWith("jdbc:")) {
+            throw new IllegalArgumentException("Invalid JDBC connection string provided: " + url);
+        }
         URI uri;
         try {
-            uri = new URI(url);
+            uri = new URI(url.substring(5));
```

Comment:
Just to clarify, `pgjdbc` does not recognize `jdbc:postgresql:User=postgres;Password=admin;Database=postgres;Server=127.0.0.1;Port=5432;` such url right?

Reply:
[pgjdbc allows the following](https://jdbc.postgresql.org/documentation/use/):

```
jdbc:postgresql:database
jdbc:postgresql:/
jdbc:postgresql://host/database
jdbc:postgresql://host/
jdbc:postgresql://host:port/database
jdbc:postgresql://host:port/
```

---

However I think the question rather is what the default behaviour is and what we want to prevent here.
If we only want to prevent pgjdbc to connect to localhost, we should allow other JDBC drivers to be able to connect to localhost instances and not fail, because the url can't be parsed in a meaningful way.

Reply:
> but I'd rather we have some false positives here that can be worked around with fdw.allow_local then to open up restricted services to users.

In our cloud environment, it's critical to avoid setting fdw.allow_local to true. Enabling this could inadvertently grant superuser privileges to all users with AL, a risk we cannot take, especially considering the possibility of future FDW implementations that might permit arbitrary queries.

Yet, it seems excessive to disrupt numerous JDBC FDW integrations by strictly requiring URLs to be formatted in a manner compatible with pgJDBC. This approach may be too restrictive.

Reply:
I think at least this should be combined with never allowing `crate` as (mapped) user for any user other than the superuser `crate`

Reply:
Cloud by default also won't have any other JDBC drivers available.
I think the fix here is good enough to unblock 5.7, we can follow up with an improvement once we've a better idea how to deal with this.

Reply:
Also created an issue that touches on this: https://github.com/crate/crate/issues/15793